### ISSUE TITLE
deterministic test: kafka producer panicked at 'called `Option::unwrap()` on a `None` value

### ISSUE THREAD

**Issue by kwannoel** [risingwavelabs/risingwave#7633]
### Describe the bug

```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/tests/simulation/src/kafka.rs:64:56
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:114:5
   3: core::option::Option<T>::unwrap
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/option.rs:823:21
   4: risingwave_simulation::kafka::producer::{{closure}}
             at ./src/tests/simulation/src/kafka.rs:64:35
   5: async_task::raw::RawTask<F,T,S>::run
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.3.0/src/raw.rs:511:20
   6: madsim::sim::task::Executor::run_all_ready
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.14/src/sim/task.rs:208:17
   7: madsim::sim::task::Executor::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.14/src/sim/task.rs:159:13
   8: madsim::sim::runtime::Runtime::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.14/src/sim/runtime/mod.rs:124:9
   9: madsim::sim::runtime::builder::Builder::run::{{closure}}::{{closure}}::{{closure}}
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/madsim-0.2.14/src/sim/runtime/builder.rs:128:35
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
note: run with `MADSIM_TEST_SEED=1` environment variable to reproduce this error
      and make sure `MADSIM_CONFIG_HASH=1AD2E4ABC41F3553`

```

### To Reproduce

Not sure.

### Additional context

Logs: https://buildkite.com/risingwavelabs/pull-request/builds/16544#018607c5-8dac-4d77-888e-36f0aa472a80

~Perhaps related? https://github.com/risingwavelabs/risingwave/issues/6931~

**Comment by wangrunji0408** [comment]
This looks weird. The panic shows that there was a file in `./scripts/source/test_data` with an invalid name (unlike `avro_bin.1`). But I don't see any problems there. 

https://github.com/risingwavelabs/risingwave/blob/1193d5370e619a2dfae385b695941754ae63d04e/src/tests/simulation/src/kafka.rs#L64

Besides, I can not reproduce this bug and it seems disappeared after the rerun. I have no idea about this.
Maybe we can improve the error message here and see if it happens again.

**Comment by kwannoel** [comment]
I guess previously kafka is not running because incorrect path, thanks for diagnosing and fixing it @wangrunji0408  🤣 

<img width="989" alt="Screenshot 2023-02-07 at 4 27 14 PM" src="https://user-images.githubusercontent.com/47273164/217191669-27b3ada0-d7d1-4e12-835b-45a34ffcf990.png">

**Comment by wangrunji0408** [comment]
Right 🤣

After printing the detailed message, I got:
```
invalid file name: "ch_benchmark". expected format "topic.partitions"
```

It turned out that a `test_data.zip` will be decompressed into the `test_data` directory, which creates a `ch_benchmark` folder.

https://github.com/risingwavelabs/risingwave/blob/1193d5370e619a2dfae385b695941754ae63d04e/ci/scripts/deterministic-e2e-test.sh#L12-L16

I'll fix this bug in #7733.
